src/thunderbots/software/ai/hl/stp/tactic/free_kick_tactic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "free_kick_tactic.h"
+
+#include "ai/hl/stp/action/kick_action.h"
+#include "ai/hl/stp/action/chip_action.h"
+#include "ai/hl/stp/evaluation/calc_best_shot.h"
+#include "ai/hl/stp/evaluation/deflect_off_enemy_target.h"
+#include "ai/hl/stp/evaluation/indirect_chip.h"
+
+#include "shared/constants.h"
+
+FreeKickTactic::FreeKickTactic(const World& world, bool loop_forever) : world(world), Tactic(loop_forever)
+{
+}
+
+std::string FreeKickTactic::getName() const
+{
+    return "Free Kick Tactic";
+}
+
+void FreeKickTactic::updateParams(const World& updated_world)
+{
+    this->world = updated_world;
+}
+
+
+double FreeKickTactic::calculateRobotCost(const Robot& robot, const World& world) 
+{
+    return dist(robot.position(), world.ball().position());
+}
+
+
+void FreeKickTactic::calculateNextIntent(IntentCoroutine::push_type& yield) 
+{
+    KickAction kick_action = KickAction();
+    Point target = world.field().enemyGoal();
+
+    do
+    {
+        // Get best shot, if any
+        auto best_shot = Evaluation::calcBestShotOnEnemyGoal(world.field(), world.friendlyTeam(), world.enemyTeam(), *robot);
```

Comment:
Probably want to make this best shot use the ball as the origin (there's a function that accept shot origin, you can use the ball position)

Reply:
fixed